### PR TITLE
[Quant][test] Add test for mixed dtypes in the same model

### PR DESCRIPTION
**Pull request by andrewor14**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #96104

Summary: This commit adds a test for mixing multiple dtypes
for different layers in the same model. The test verifies that
FX graph mode quantization converts the dtypes correctly
between the layers.

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_mixed_dtypes

Reviewers: jcaip, vkuzo, supriyar

Subscribers: jcaip, vkuzo, supriyar

### CODE REVIEW

File: test/quantization/fx/test_quantize_fx.py [hunk 1]
```diff
@@ -5975,6 +5975,118 @@ def forward(self, x):
...
+        )
+        fixed_qparams_op_qint32_dtype_config = DTypeConfig(
+            input_dtype=torch.qint32,
+            output_dtype=torch.qint32,
+        )
+        backend_config = get_qnnpack_backend_config()
+        for config in backend_config.configs:
+            if config.pattern == torch.nn.Linear:
+                config.add_dtype_config(weighted_op_qint32_dtype_config)
+            elif config.pattern in [torch.nn.Sigmoid, torch.nn.Tanh]:
```
Comment (jcaip) [inline test/quantization/fx/test_quantize_fx.py:6033]: double checking, do you mean to add both when you see either Sigmoid and Tanh? or to add them respectively.

**Review by jcaip — APPROVED** [review]
lgtm

File: test/quantization/fx/test_quantize_fx.py (hunk 1 above)
Comment (andrewor14) [reply test/quantization/fx/test_quantize_fx.py:6033]: Yeah right now this supports both dtypes for both ops. I guess it can be more specific but I think this is also fine

**Comment by andrewor14** [comment]
Thanks, merging this
